### PR TITLE
Dispatcher: add File browser actions 'Set display mode', 'Sort By'

### PR DESCRIPTION
**Pull request by hius07**
Closes https://github.com/koreader/koreader/issues/11007.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11921)

### CODE REVIEW

**Comment by poire-z** [comment]
I guess this (slightly?) conflict with #11792, if we go on with it.
But it seems we could rewrite a small bit of it (reintroducing the shortened `local modes` in `CoverBrowser:addToMainMenu(menu_items)`) and it wouldn't hurt.

**Comment by hius07** [comment]
I can postpone until merging https://github.com/koreader/koreader/pull/11792, and then adapt this one.

**Comment by poire-z** [comment]
Well, none of us will probably work on #11792 anytime soon.
And this PR shows the current "model" of settings (https://github.com/koreader/koreader/pull/11792#issuecomment-2118248756) can be better in some cases than the natural one of #11792 - you would be quite embarassed having to add 2 other actions to show cover or show filename only.

I'd say: merge this one, and we will adapt #11792 when we have more time to think about it.

**Comment by hius07** [comment]
> you would be quite embarassed having to add 2 other actions to show cover or show filename only

We might need a proxy-method transforming current 5 modes into the set of 3(?) new settings and vice versa.

**Comment by Frenzie** [comment]
> Closes #11007.

I think you also have to edit it into the OP.

File: frontend/dispatcher.lua [hunk 1]
```diff
@@ -121,12 +122,17 @@ local settingsList = {
     ----
 
     -- File browser
+    set_display_mode = {category="string", event="SetDisplayMode", title=_("Set display mode"), args_func=FileManager.getDisplayModeActions, filemanager=true},
```
Comment (poire-z) [inline frontend/dispatcher.lua:125]: (There are a few other existing args_func and stuff name something**Actions**. Any idea why we named that with **Actions** ? It sounds a bit odd we named these like thins. In general, we name somethingAction the little functions we create to give to `UIManager:scheduleIn(1, do_something_action)`)

**Review by poire-z — APPROVED** [review]

File: frontend/dispatcher.lua (hunk 1 above)
Comment (Frenzie) [reply frontend/dispatcher.lua:125]: getDisplayModes?

**Pull request merged**